virtualizarr/tests/test_backend.py

```diff
@@ -438,3 +440,114 @@ def test_open_dataset_with_scalar(self, hdf5_scalar, hdf_backend):
+        vds1 = open_virtual_dataset(filepath1, indexes={})
+        vds2 = open_virtual_dataset(filepath2, indexes={})
+        expected_vds = xr.concat(
+            [vds1, vds2], dim="time", coords="minimal", compat="override"
+        )
+        xrt.assert_identical(combined_vds, expected_vds)
+
+        # test combine by coords using in-memory indexes
+        combined_vds = open_virtual_mfdataset(
+            [filepath1, filepath2], combine="by_coords", loadable_variables=["time"]
+        )
+        vds1 = open_virtual_dataset(filepath1, loadable_variables=["time"])
+        vds2 = open_virtual_dataset(filepath2, loadable_variables=["time"])
+        expected_vds = xr.concat(
+            [vds1, vds2], dim="time", coords="minimal", compat="override"
+        )
+        xrt.assert_identical(combined_vds, expected_vds)
+
+        # test combine by coords again using in-memory indexes but for a glob
+        file_glob = Path(filepath1).parent.glob("air*.nc")
+        combined_vds = open_virtual_mfdataset(
+            file_glob, combine="by_coords", loadable_variables=["time"]
+        )
+        xrt.assert_identical(combined_vds, expected_vds)
+
+    # @requires_dask
+    def test_dask(self, netcdf4_files_factory): ...
+
+    @requires_lithops
+    def test_lithops(self, netcdf4_files_factory):
+        # by default this will use the lithops LocalHost executor
+
+        filepath1, filepath2 = netcdf4_files_factory()
+
+        # test combine nested without in-memory indexes
+        combined_vds = open_virtual_mfdataset(
+            [filepath1, filepath2],
+            combine="nested",
+            concat_dim="time",
+            coords="minimal",
+            compat="override",
+            indexes={},
+            parallel="lithops",
+        )
+        vds1 = open_virtual_dataset(filepath1, indexes={})
+        vds2 = open_virtual_dataset(filepath2, indexes={})
+        expected_vds = xr.concat(
+            [vds1, vds2], dim="time", coords="minimal", compat="override"
+        )
+
+        print(combined_vds)
+        print(expected_vds)
+        print(combined_vds.indexes)
+        print(combined_vds.indexes)
+        print(combined_vds["lat"].attrs)
+        print(expected_vds["lat"].attrs)
+        print(combined_vds["lat"].encoding)
+        print(expected_vds["lat"].encoding)
+        print(combined_vds["lat"].data)
+        print(expected_vds["lat"].data)
+        print(combined_vds["lat"].data.zarray)
+        print(expected_vds["lat"].data.zarray)
+        print(combined_vds["lat"].data.manifest.dict())
+        print(expected_vds["lat"].data.manifest.dict())
+
+        # TODO this assertion unintentially triggers loading, see issue #354
+        # xrt.assert_identical(combined_vds.coords.variables['lat'], expected_vds.coords.variables['lat'])
+
+        # TODO I have no idea why this assertion fails for all the coords - everything about the coords looks identical
+        # xrt.assert_identical(combined_vds, expected_vds)
```

Comment:
I'm stuck on why this `assert_identical` on the datasets doesn't pass. It complains that all the coordinates are different, but every attribute I can think to check looks identical 😖